src/main/java/io/netty/buffer/api/Component.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api;
+
+import java.nio.ByteBuffer;
+import java.util.function.Consumer;
+
+/**
+ * A view onto the buffer component being processed in a given iteration of {@link Buf#forEachReadable(Consumer)}.
+ * <p>
+ * Instances of this interface are allowed to be mutable behind the scenes, and the data is only guaranteed to be
+ * consistent within the given iteration.
+ */
+public interface Component {
+
+    /**
+     * Check if this component is backed by a cached byte array than can be accessed cheaply.
```

Comment:
I think I would also state the the contract is the same as `array` in the sense that this is not meant to be written too.